### DISCUSSION TITLE
yolo cooamnd

### DISCUSSION THREAD

**Discussion by tejasw20** [ultralytics/ultralytics#5754] [Q&A]
!yolo task=detect mode=predict model=/home/Downloads/best.onnx  imgsz=416 hide_labels=True, hide_conf=True save_txt=True save_crop=True source= '/home/Downloads/Dataset/train/images'
How to used this command directly on python code? anyone  have idea please provide the answer?

**Comment by ahillsavio1** [comment]
Try this:

```
from ultralytics import YOLO

model = YOLO('/home/Downloads/best.onnx')
results = model.predict(
    imgsz=416,
    hide_labels=True,
    hide_conf=True,
    save_txt=True,
    save_crop=True,
    source= '/home/Downloads/Dataset/train/images'
)
```

Do let me know if it worked out.

Thank you.

**Reply by tejasw20** [reply]
thank you for answer

**Reply by tejasw20** [reply]
but i want to use this cli  !yolo task=detect mode=predict model=/home/Downloads/best.onnx imgsz=416 hide_labels=True, hide_conf=True save_txt=True save_crop=True source= '/home/Downloads/Dataset/train/images' 
command directly  on my python code is this possible??

**Reply by ahillsavio1** [reply]
Isn't that what I answered to in the first place?

**Reply by tejasw20** [reply]
no

**Reply by glenn-jocher** [reply]
@tejasw20 apologies for any confusion. To use the CLI command directly in your Python code, you can use the `subprocess` module to call the command as you would in a terminal. Here's a concise example of how you might do it:

```python
import subprocess

command = "yolo task=detect mode=predict model=/home/Downloads/best.onnx imgsz=416 hide_labels=True hide_conf=True save_txt=True save_crop=True source='/home/Downloads/Dataset/train/images'"
subprocess.run(command, shell=True)
```

Make sure to remove the exclamation mark `!` at the beginning of the command when using it in Python with `subprocess`. The `shell=True` parameter allows you to run the command as if it were in a shell terminal.
